src/lib/sw_engine/tvgSwRaster.cpp

```diff
@@ -113,6 +113,24 @@ static uint32_t _applyBilinearInterpolation(const uint32_t *img, uint32_t w, uin
     return COLOR_INTERPOLATE(COLOR_INTERPOLATE(c1, 255 - dX, c2, dX), 255 - dY, COLOR_INTERPOLATE(c4, 255 - dX, c3, dX), dY);
 }
 
+static uint32_t _average2Nx2NPixel(SwSurface* surface, const uint32_t *img, uint32_t w, uint32_t h, uint32_t rX, uint32_t rY, uint32_t n)
+{
+    uint32_t c[4] = { 0 };
+    auto n2 = n * n;
+
+    for (auto x = rX - n; x < rX + n; ++x) {
+        for (auto y = rY - n; y < rY + n; ++y) {
+            c[0] += img[x + (y * w)] >> 24;
+            c[1] += (img[x + (y * w)] >> 16) & 0xff;
+            c[2] += (img[x + (y * w)] >> 8) & 0xff;
+            c[3] += img[x + (y * w)] & 0xff;
```

Comment:
I'd write ```auto pix = img[x + (y * w)]``` and then use `pix` to avoid 4 multiplics

Reply:
modified